Data/Aeson/Parser/Internal.hs

```diff
@@ -322,7 +322,7 @@ jstring_ = do
   -- not sure whether >= or bit hackery is faster
   -- perfectly, we shouldn't care, it's compiler job.
   s <- A.takeWhile (\w -> w /= DOUBLE_QUOTE && w /= BACKSLASH && w >= 0x20 && w < 0x80)
-  let txt = TE.decodeUtf8 s
+  let txt = TE.decodeLatin1 s
```

Comment:
could you make a local definition, and name it `unsafeDecodeAscii` with a comment that

```
-- | The input is assumed to contain only 7bit ASCII characters (i.e. @< 0x80@).
unsafeDecodeAscii :: ByteString -> Text
```

That name will communicate better what we try to do. In fact, I think we can win more with custom FFI (`+cffi`) i.e. do what `decodeLatin1` does, but with even less checks (as we have checked that there is no `0x80` bytes).

Perfectly, `unsafeDecodeAscii` would be in `text`.

Reply:
I took a stab at accelerating `decodeLatin1` https://github.com/haskell/text/pull/297